modules/cudf/src/data_frame.ts

```diff
@@ -253,13 +264,75 @@ export class DataFrame<T extends TypeMap = any> {
    */
   filter(mask: Series<Bool8>) {
     const temp       = new Table({columns: this._accessor.columns});
-    const columns    = temp.gather(mask._col);
+    const columns    = temp.gather(mask._col, false);
     const series_map = {} as SeriesMap<T>;
     this._accessor.names.forEach(
       (name, index) => { series_map[name] = Series.new(columns.getColumnByIndex(index)); });
     return new DataFrame(series_map);
   }
 
+  /**
+   * Join columns with other DataFrame.
+   *
+   * @param props the configuration for the join
+   * @returns the joined DataFrame
+   */
+  join<U extends TypeMap, R extends keyof U>(props: JoinProps<U, R>): DataFrame<any> {
```

Comment:
Definitely welcome any input on if/how the return type here might be improved.

Reply:
`DataFrame<T&U>`?

Reply:
ah understood